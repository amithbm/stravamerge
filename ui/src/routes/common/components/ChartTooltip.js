import React from 'react'
import { Panel } from 'react-bootstrap'

const ChartTooltip = ({payload, label}) => {
    return <Panel>
            {label}: {payload[0].value}km
        </Panel>
}

ChartTooltip.propTypes = {
    label: React.PropTypes.string,
    payload: React.PropTypes.array
}

export default ChartTooltip
