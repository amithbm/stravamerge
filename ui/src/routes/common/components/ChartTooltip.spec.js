import React from 'react'
import {shallow} from 'enzyme'
import ChartTooltip from './ChartTooltip'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<ChartTooltip />', () => {
        it('renders as expected', () => {
            const component =  shallow(<ChartTooltip label="hello world" payload={[ { value: 10 }]} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})

