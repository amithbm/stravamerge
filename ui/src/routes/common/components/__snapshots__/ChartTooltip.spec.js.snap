exports[`components <ChartTooltip /> renders as expected 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  defaultExpanded={false}>
  hello world
  : 
  10
  km
</Panel>
`;
